src/js/reducers/conversation-reducer.js

```diff
@@ -118,25 +127,28 @@ export function ConversationReducer(state = INITIAL_STATE, action) {
             };
         case ConversationActions.SET_CONVERSATION:
             return {
-                ...action.conversation,
-                messages: state.messages
+                ...INITIAL_STATE,
+                ...action.conversation
             };
         case ConversationActions.SET_MESSAGES:
             return {
                 ...state,
-                messages: assignGroups(sortMessages(removeDuplicates(action.messages)))
+                messages: assignGroups(sortMessages(cleanUpMessages(action.messages))),
+                quickReplies: extractQuickReplies(action.messages[action.messages.length - 1])
```

Comment:
That's how it was working before. I changed it because we want to not show messages with replies and no text or image. I started by just omitting them at render time. Then, grouping logic was off, because it considered empty messages in there. So I tried to tweak the logic to ignore empty messages aaaaaaaaaaaaand it was a nightmare with so many edge cases. 

The simple solution was to extract the quick replies from the last message, store them there, and discard any empty message so the grouping and rendering logic wouldn't even need to be changed.